common/pom.xml

```diff
@@ -37,11 +37,21 @@
       <artifactId>drill-protocol</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter-engine</artifactId>
+      <version>${junit.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.junit.vintage</groupId>
+      <artifactId>junit-vintage-engine</artifactId>
+      <version>${junit.version}</version>
+    </dependency>
     <dependency>
       <!-- add as provided scope so that we can compile TestTools.  Should only be ever used in a test scenario where someone else is bringing JUnit in. -->
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
-      <version>${junit.version}</version>
+      <scope>provided</scope>
```

Comment:
Please remove junit / jupiter dependencies from here and move `RepeatTestRule` to the tests folder.

Reply:
@vvysotskyi RepeatTestRule is currently at common/src/main/java/org/apache/drill/common/util/RepeatTestRule.java
But it is used by:
1) exec/java-exec/src/test/java/org/apache/drill/exec/server/TestDrillbitResilience.java
2) common/src/test/java/org/apache/drill/test/TestTools.java

So, to do what you suggest I have to copy it in two modules - java-exec and common.
Or to move it in common/src/test/... and then in java-exec/pom.xml to add a dependency to common-tests.jar

Reply:
`exec` module is the parent for `java-exec` and it contains the dependency to `drill-common` with `tests` classifier, so it should be enaugh just to move `common/src/main/java/org/apache/drill/common/util/RepeatTestRule.java` class to `common/test/main/java/org/apache/drill/common/util/RepeatTestRule.java`.

Reply:
Done!